docs/spec/relayer/Relayer.tla

```diff
@@ -267,478 +204,68 @@ PendingDatagrams(srcChainID, dstChainID) ==
  ***************************************************************************)   
 \* Update the height of the relayer client for some chainID
 UpdateRelayerClients(chainID) ==
-        /\ relayerChainHeights[chainID] < GetLatestHeight(chainID)
-        /\ relayerChainHeights' = [relayerChainHeights EXCEPT 
-                                        ![chainID] = GetLatestHeight(chainID)
+        /\ relayerHeights[chainID] < GetLatestHeight(GetChainByID(chainID))
+        /\ relayerHeights' = [relayerHeights EXCEPT 
+                                        ![chainID] = GetLatestHeight(GetChainByID(chainID))
                                    ]
-        /\ UNCHANGED <<chains, outgoingDatagrams>>  
+        /\ UNCHANGED <<chainAstore, chainBstore, outgoingDatagrams>>  
 
 \* for two chains, srcChainID and dstChainID, where srcChainID /= dstChainID, 
 \* create the pending datagrams and update the corresponding sets of pending datagrams
 Relay(srcChainID, dstChainID) ==
-    LET datagramsAndRelayerUpdate == PendingDatagrams(srcChainID, dstChainID) IN
+    LET datagramsAndRelayerUpdate == ComputeDatagrams(srcChainID, dstChainID) IN
     /\ srcChainID /= dstChainID
     /\ outgoingDatagrams' = 
             [outgoingDatagrams EXCEPT 
                 ![dstChainID] = outgoingDatagrams[dstChainID] 
                                 \union 
                                 datagramsAndRelayerUpdate.datagrams
             ]
-    /\ relayerChainHeights' = datagramsAndRelayerUpdate.relayerUpdate       
-    /\ UNCHANGED chains
+    /\ relayerHeights' = datagramsAndRelayerUpdate.relayerUpdate       
+    /\ UNCHANGED <<chainAstore, chainBstore>>
 
-(***************************************************************************
- Component actions
- ***************************************************************************)
-\* Relayer
-\*    The relayer either
-\*        - updates its clients, or
-\*        - relays datagrams between two chains, or
-\*        - does nothing
-Relayer ==
-    \/ \E chainID \in ChainIDs : UpdateRelayerClients(chainID)
-    \/ \E srcChainID \in ChainIDs : \E dstChainID \in ChainIDs : Relay(srcChainID, dstChainID)
-    \/ UNCHANGED vars
+UpdateRelayer ==
+    \E chainID \in ChainIDs : UpdateRelayerClients(chainID)
     
-\* Environment
-\*    The environment takes the action Env!Next
-\*    and leaves the relayer variable unchanged
-Environment ==
-    /\ Env!Next
-    /\ UNCHANGED relayerChainHeights    
+CreateDatagrams ==
+    \E srcChainID \in ChainIDs : \E dstChainID \in ChainIDs : Relay(srcChainID, dstChainID)    
```

Comment:
Can it happen that chosen srcChainID and dstChainID are the same, so Relay action does not trigger? I am wondering if it's simpler just picking srcChainID as we then know dstChainID, at least in a model with two chains.